internal/connection/connection.go

```diff
@@ -102,6 +224,16 @@ func (c *Connection) fetchCapabilities(ctx context.Context) error {
 	return nil
 }
 
+// ensureConnected re-eastablishes a connection if it is in a disconnected
+// state. An error is returned if it fails to renew the connection.
+func (c *Connection) ensureConnected() error {
+	if !c.connected {
+		return c.Connect()
+	}
+
+	return nil
+}
```

Comment:
There is very slight possibility that c.connected maybe set to false after it is checked here.

Any checks on c.connected need to be performed with mutex locks held.

I think just calling c.Connect() directly would be sufficient since you have
`if c.connected{}` within it already in line 87,

Reply:
Done